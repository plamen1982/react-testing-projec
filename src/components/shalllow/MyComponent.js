import React, { useState } from 'react';

const MyComponent = ({ strings, handleClick, isHidden }) => {
  return (
    <>
      <div>My Component for shallow testing </div>
      <div>String 1: {strings[0]}</div>
      {!isHidden && <div>String 2: {strings[1]}</div>}
      <button onClick={handleClick} id="my-button-one">
        click
      </button>
    </>
  );
};

const wrapper = () => {
  const [isHidden, setIsHidden] = useState(false);
  const strings = ['one', 'two'];
  const handleClick = () => {
    setIsHidden(true);
  };
  return (
    <MyComponent
      strings={strings}
      handleClick={handleClick}
      isHidden={isHidden}
    />
  );
};

export default MyComponent;
