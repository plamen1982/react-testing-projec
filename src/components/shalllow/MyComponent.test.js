import { mount, shallow, render } from 'enzyme';
import MyComponent from './MyComponent';
import React from 'react';

describe('shollow rendering | <MyComponent> shallow rendering', () => {
  test('shallow |<MyComponent />', () => {
    const strings = ['first', 'second'];
    const component = shallow(<MyComponent strings={strings} />);
    expect(component).toMatchSnapshot();
  });
});

describe('shallow rendering with disappear| <MyComponent> mount rendering', () => {
  test('shallow |<MyComponent /> should click the button', () => {
    const strings = ['first', 'second'];
    let isHidden = true;
    const handleClick = jest.fn();
    const component = shallow(
      <MyComponent
        strings={strings}
        isHidden={isHidden}
        handleClick={handleClick}
      />
    );
    component.find('button#my-button-one').simulate('click');
    expect(component).toMatchSnapshot();
    expect(handleClick).toHaveBeenCalled();
  });
});

describe('mount- rendering with event mocked | <MyComponent> shallow rendering', () => {
  const clickFn = jest.fn();
  test('mount- |<MyComponent /> should mocked handleClick', () => {
    const strings = ['first', 'second'];
    const component = mount(
      <MyComponent strings={strings} handleClick={clickFn} />
    );
    component.find('button#my-button-one').simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});
