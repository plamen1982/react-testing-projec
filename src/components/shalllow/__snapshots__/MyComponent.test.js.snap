// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow rendering with disappear| <MyComponent> mount rendering shallow |<MyComponent /> should click the button 1`] = `
<Fragment>
  <div>
    My Component for shallow testing 
  </div>
  <div>
    String 1: 
    first
  </div>
  <button
    id="my-button-one"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  >
    click
  </button>
</Fragment>
`;

exports[`shollow rendering | <MyComponent> shallow rendering shallow |<MyComponent /> 1`] = `
<Fragment>
  <div>
    My Component for shallow testing 
  </div>
  <div>
    String 1: 
    first
  </div>
  <div>
    String 2: 
    second
  </div>
  <button
    id="my-button-one"
  >
    click
  </button>
</Fragment>
`;
